batch/test/advanced_creation.js

```diff
@@ -0,0 +1,276 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const path = require('path');
+const cp = require('child_process');
+const {describe, it, before} = require('mocha');
+const {BatchServiceClient} = require('@google-cloud/batch').v1;
+const batchClient = new BatchServiceClient();
+const {Storage} = require('@google-cloud/storage');
+const storage = new Storage();
+const {ProjectsClient} = require('@google-cloud/resource-manager').v3;
+const resourcemanagerClient = new ProjectsClient();
+const compute = require('@google-cloud/compute');
+const instanceTemplatesClient = new compute.InstanceTemplatesClient();
+
+// get a short ID for this test run that only contains characters that are valid in UUID
+// (a plain UUID won't do because we want the "test-job-js" prefix and that would exceed the length limit)
+const {customAlphabet} = require('nanoid');
+const allowedChars = 'qwertyuiopasdfghjklzxcvbnm';
+const testRunId = customAlphabet(allowedChars, allowedChars.length)();
+
+const execSync = cmd => cp.execSync(cmd, {encoding: 'utf-8'});
+
+const cwd = path.join(__dirname, '..');
+
+describe('Create jobs with container, template and bucket', () => {
+  let projectId;
+  let bucketName;
+  let templateName;
+
+  before(async () => {
+    projectId = await batchClient.getProjectId();
+    bucketName = `batch-js-test-bucket-${testRunId}`;
+    templateName = `batch-js-test-template-${testRunId}`;
+    await createBucket(bucketName);
+    await createTemplate(projectId, templateName);
+  });
+
+  it('create a job with a container payload', async () => {
+    execSync(
```

Comment:
Are we catching errors from these and/or checking the sample output? We might also not want to use `async` so that exceptions flow up to mocha, since there are no `await` statements in there.

Reply:
Also I'm being _incredibly_ nitpicky here :) but usually you'd word it like `it('creates a job...')` so that the mocha output flows well.